torchinductor/codegen/cpp.py

```diff
@@ -321,14 +326,7 @@ def store(self, name, index, value, mode=None):
             if config.cpp.threads == 1:
                 line = f"{var}[{cexpr(index)}] += {value};"
             else:
-                # Note atomic_ref requires C++20 and certain processor features
-                self.stores.writeline(
-                    name,
-                    "static_assert(std::atomic_ref<"
-                    + f"std::remove_pointer_t<decltype({var})>"
-                    + ">::is_always_lock_free);",
-                )
-                line = f"std::atomic_ref({var}[{cexpr(index)}]) += {value};"
+                line = f"cpu_atomic_add_float(&{var}[{cexpr(index)}], {value});"
```

Comment:
a memory layout and sizeof() assert wouldn't hurt here, or upstreamed into cpu_atomic_add_float as its a little loosey-goosey around some edge cases between the input types AND has float assumptions. If we ever need more than floats here (maybe we do already?) we'll need to wrap this into a template and invoke the right one, and potentially provide our own for int.